src/ott/problems/nn/dataset.py

```diff
@@ -0,0 +1,155 @@
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+__all__ = ['create', 'Dataloaders', 'GaussianMixture']
+
+Name_t = Literal['gaussian_simple', 'gaussian_circle', 'gaussian_square_five',
+                 'gaussian_square_four']
+
+
+class Dataloaders(NamedTuple):
+  r"""Train and validation dataloaders for the source and target measures.
+
+  Args:
+  trainloader_source: Training dataset, source measure
+  trainloader_target: Training dataset, target measure
+  validloader_source: Valid dataset, source measure
+  validloader_target: Valid dataset, target measure
+  """
+  trainloader_source: Iterable[jnp.ndarray]
+  trainloader_target: Iterable[jnp.ndarray]
+  validloader_source: Iterable[jnp.ndarray]
+  validloader_target: Iterable[jnp.ndarray]
+
+
+GAUSSIAN_CENTERS = {
+    'simple':
+        np.array([[0, 0]]),
+    'circle':
+        np.array([
+            (1, 0),
+            (-1, 0),
+            (0, 1),
+            (0, -1),
+            (1.0 / np.sqrt(2), 1.0 / np.sqrt(2)),
+            (1.0 / np.sqrt(2), -1.0 / np.sqrt(2)),
+            (-1.0 / np.sqrt(2), 1.0 / np.sqrt(2)),
+            (-1.0 / np.sqrt(2), -1.0 / np.sqrt(2)),
+        ]),
+    'square_five':
+        np.array([[0, 0], [1, 1], [-1, 1], [-1, -1], [1, -1]]),
+    'square_four':
+        np.array([[1, 0], [0, 1], [-1, 0], [0, -1]]),
+}
+
+
+@dataclasses.dataclass
+class GaussianMixture:
+  """A mixture of Gaussians.
+
+  Args:
+    name: the name specifying the centers of the mixture components;
+      `simple` (data clustered in one center),
+      `circle` (two-dimensional Gaussians arranged on a circle),
+      `square_five` (two-dimensional Gaussians on a square with one Gaussian
+      in the center), and `square_four` (two-dimensional Gaussians in the
+      corners of a rectangle)
+    batch_size: batch size of the samples
+    init_key: initial PRNG key
+    scale: scale of the individual Gaussian samples
+    variance the variance of the individual Gaussian samples
+  """
+  name: str
+  batch_size: int
+  init_key: jax.random.PRNGKey
+  scale: float = 5.0
+  variance: float = 0.5
+
+  def __post_init__(self):
+    if self.name not in GAUSSIAN_CENTERS:
+      raise ValueError(
+          f'{self.name} is not a valid dataset for GaussianMixture'
+      )
+    self.centers = GAUSSIAN_CENTERS[self.name]
+
+  def __iter__(self):
+    return self.create_sample_generators()
+
+  def create_sample_generators(self) -> Iterator[jnp.array]:
+    # create generator which randomly picks center and adds noise
+    key = self.init_key
+    while True:
+      k1, k2, key = jax.random.split(key, 3)
+      means = jax.random.choice(k1, self.centers, [self.batch_size])
+      normal_samples = jax.random.normal(k2, [self.batch_size, 2])
+      samples = self.scale * means + self.variance ** 2 * normal_samples
+      yield samples
+
+
+def get_sampler(name: str, **kwargs: Any) -> Iterable[jnp.ndarray]:
```

Comment:
I don't think we need this function, `Name_t` can be `Literal['gaussian_square', 'gaussian_circle', ...]`.

Reply:
I removed the `gaussian_` prefix from here for now since they're all Gaussian and inlined `get_sampler` back in to get rid of it. Are you saying to remove `Name_t` as well?

Reply:
No, the `Name_t` alias is good, thank you!